### ISSUE TITLE
Font expansion workflow

### ISSUE THREAD

**Issue by wboykinm** [mapbox/mapbox-studio-classic#395]
It'd be nice to have some guidance on how we might request or pull-request system fonts to make them available on Mapbox-hosted TM2 packages. Obviously 90% of this ticket relates to the internal Mapbox server system, but at some point I'd like to see the options expand to include the "maybe"s on [this list](https://github.com/mapbox/tm2/issues/8). Happy to fork and pull, but again I know that TM2 defaults are the least of the hurdles between seeing a font in TM2 and getting it served at a Mapbox endpoint.

Thanks!

**Comment by yhahn** [comment]
@springmeyer is working on a way that users can include their own fonts with uploaded styles which will be a more sustainable solution than a fixed/curated list : ) Will keep you posted.

**Comment by nilshoerrmann** [comment]
It would be interesting if I could integrate Typekit-Fonts in a project by specifying the kit ID. Not sure if it's technically possible though given that tiles are processed by Mapnik and not the browser.

**Comment by rossmeissl** [comment]
Typekit (via kit ID) would be perfect :+1:

**Comment by wboykinm** [comment]
@springmeyer @yhahn Any word on this? Any chance [this talk of sprites](https://github.com/mapbox/tm2/issues/498) is related? 

Don't underestimate how much of the preprocessing (sprite generation, say) users would be willing to do to make custom fonts available :)

**Comment by yhahn** [comment]
Yes, "coming soon"

**Comment by wboykinm** [comment]
Will now see air quotes whenever fonts come up . . .

**Comment by yhahn** [comment]
Yes, "soon" use of "custom" "fonts" will be "available" in your "styles" especially if you send "@springmeyer" "craft" "beer"

**Comment by wboykinm** [comment]
Waiting for drone delivery to be available. The man lives remote.

**Comment by nerik** [comment]
We'll have to wait for the FAA's green light apparently?
http://techcrunch.com/2014/02/04/the-faa-shuts-down-beer-delivery-drone/

In the meantime @springmeyer, if you want to taste our excellent 'Bière de l'Ourcq', please give me your address. I'm serious about this.

**Comment by wboykinm** [comment]
@yhahn @springmeyer Is there a roadmap for bringing [all of the fontfont](https://www.fontfont.com/fonts#utf8=%E2%9C%93&weight=250%2C1000&width=1%2C10&intended_use=All&sortorder=2) families online? I just pulled the latest build and I only (!) see ~100 of them.

And Dane is getting some Heady Topper the next time he wanders east.

**Comment by springmeyer** [comment]
@wboykinm no plans that I'm aware of to bring more online. As far as custom fonts in general I'm closing in on support and will update this ticket hopefully by the end of this week with details.

**Comment by KP** [comment]
Super excited about custom fonts! Any update on when this may be possible?

**Comment by MichaelMarkieta** [comment]
Would love to hear an update.

**Comment by springmeyer** [comment]
The main puzzle piece to get this working is mostly done in Mapnik 3.x, but is waiting on a node-mapnik release based on Mapnik 3.x which involved a lot more that just font improvements. Thanks for everyone's patience. You can see when this lands at https://github.com/mapnik/node-mapnik/issues/317.

**Comment by springmeyer** [comment]
Mapnik 3.x is now deployed and upcoming Mapbox Studio (v0.1.7) will support custom fonts (see #973 and https://github.com/mapbox/mapbox-studio/blob/mb-pages/docs/0000-01-01-advanced-features.md). NOTE: v0.1.7 will arrive once we can fix the startup crasher in the Windows build.

**Comment by wboykinm** [comment]
:beers:
